web/courses_test.go

```diff
@@ -484,67 +504,38 @@ func TestPromoteDemoteRejectTeacher(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if _, err := ags.UpdateEnrollment(ctx, &pb.Enrollment{
-		UserID:   student1.ID,
-		CourseID: course.ID,
-		Status:   pb.Enrollment_TEACHER,
-	}); err != nil {
-		t.Fatal(err)
-	}
-
-	if _, err := ags.UpdateEnrollment(ctx, &pb.Enrollment{
-		UserID:   student2.ID,
-		CourseID: course.ID,
-		Status:   pb.Enrollment_TEACHER,
-	}); err != nil {
-		t.Fatal(err)
-	}
-
-	// promote the TA to teacher as well
-	if _, err := ags.UpdateEnrollment(ctx, &pb.Enrollment{
-		UserID:   ta.ID,
-		CourseID: course.ID,
-		Status:   pb.Enrollment_TEACHER,
-	}); err != nil {
+	request.Enrollments = []*pb.Enrollment{student1_enrollment, student2_enrollment, ta_enrollment}
+	if _, err := ags.UpdateEnrollments(ctx, request); err != nil {
 		t.Fatal(err)
 	}
 
 	// TA attempts to demote self, must succeed
+	ta_enrollment.Status = pb.Enrollment_STUDENT
+	request.Enrollments = []*pb.Enrollment{ta_enrollment}
 	ctx = qtest.WithUserContext(context.Background(), ta)
-	if _, err := ags.UpdateEnrollment(ctx, &pb.Enrollment{
-		UserID:   ta.ID,
-		CourseID: course.ID,
-		Status:   pb.Enrollment_STUDENT,
-	}); err != nil {
+	if _, err := ags.UpdateEnrollments(ctx, request); err != nil {
 		t.Fatal(err)
 	}
 
 	// student2 attempts to demote course creator, must fail
+	teacher_enrollment.Status = pb.Enrollment_STUDENT
+	request.Enrollments = []*pb.Enrollment{teacher_enrollment}
 	ctx = qtest.WithUserContext(context.Background(), student2)
-	if _, err := ags.UpdateEnrollment(ctx, &pb.Enrollment{
-		UserID:   teacher.ID,
-		CourseID: course.ID,
-		Status:   pb.Enrollment_STUDENT,
-	}); err == nil {
-		t.Error("expected error: 'only course creator can change status of other teachers'")
+	if _, err := ags.UpdateEnrollments(ctx, request); err == nil {
+		t.Error("expected error: 'only course creator can change status of other teachers'", err)
 	}
 
 	// student2 attempts to reject course creator, must fail
-	if _, err := ags.UpdateEnrollment(ctx, &pb.Enrollment{
-		UserID:   teacher.ID,
-		CourseID: course.ID,
-		Status:   pb.Enrollment_NONE,
-	}); err == nil {
+	teacher_enrollment.Status = pb.Enrollment_NONE
+	if _, err := ags.UpdateEnrollments(ctx, request); err == nil {
 		t.Error("expected error: 'only course creator can change status of other teachers'")
 	}
 
 	// teacher demotes student1, must succeed
+	student1_enrollment.Status = pb.Enrollment_STUDENT
```

Comment:
In Go we don't like underscore ;-) `student1Enrollment` and `teacherEnrollment` should do it...